ibis/mapd/client.py

```diff
@@ -448,6 +471,12 @@ def _get_schema_using_query(self, query):
 
         return sch.Schema(names, ibis_types)
 
+    def _get_schema_using_validator(self, query):
+        result = self.con._client.sql_validate(self.con._session, query)
+        return sch.Schema.from_dict({r: MapDDataType._mapd_to_ibis_dtypes[
```

Comment:
Use `sch.Schema.from_tuples` instead of `from_dict` so that order of columns is preserved.

Reply:
yes!

Reply:
Working testing.  I'm still a little uncertain about how ibis testing is dispatched, and what specific testing criteria would be appropriate for this change.

Reply:
@mpeaton maybe you can improve this https://github.com/ibis-project/ibis/blob/master/ibis/tests/all/test_client.py#L69

also if you want to add tests that could break in another backend .. you can also add tests for `sql` at https://github.com/ibis-project/ibis/blob/master/ibis/mapd/tests/test_client.py

if you want any help related to tests ping me :)

Reply:
After some deliberation, it seems that _get_schema already has test coverage within ibis/tests/all/test_client.py::test_query_schema

Reply:
For this particular PR at least...

Reply:
sure raw SQLs would be much better!